server/service/transport_error.go

```diff
@@ -132,6 +132,8 @@ func encodeError(ctx context.Context, err error, w http.ResponseWriter) {
 		if e.NodeInvalid() {
 			w.WriteHeader(http.StatusUnauthorized)
 			errMap["node_invalid"] = true
+		} else if e.statusCode != 0 {
+			w.WriteHeader(e.statusCode)
```

Comment:
we've been making responses implement a `Status` method to accomplish this, since we do this here:

https://github.com/fleetdm/fleet/blob/ca435eb2444d0cb95d7fec6fb94cea46af73ec41/server/service/transport.go#L47-L52

is it possible to do the same here? for consistency

Reply:
yes, updated!